src/textual/geometry.py

```diff
@@ -117,7 +123,15 @@ def contains_point(self, point: tuple[int, int]) -> bool:
         width, height = self
         return width > x >= 0 and height > y >= 0
 
-    def __contains__(self, other: Any) -> bool:
+    def __contains__(self, other: Iterable[int]) -> bool:
```

Comment:
Have you checked this with Mypy? (`make typecheck`). These methods accept any argument, which is why they are typed so broadly.

Reply:
Yes, MyPy correctly flags this as an LSP violation -- my bad. Should `__contains__` raise a TypeError, though, if it accepts an argument of any type? To be consistent with the type signature, perhaps it should return `False` rather than raising an exception?

Reply:
Good point. A container wouldn't raise TypeError for anything. But you could argue that this class isn't really a container per se, the concept of being "in" something means something different than a dict for example.

My gut feeling is that the TypeError may catch bugs, so maybe it is best left in. Unless you can think of a reason not to?

Reply:
Yes, I think you're right -- strictly speaking it might break LSP, but it's probably better to leave the `TypeError` in. Would you prefer that I change the annotation back to `other: Any` (in the knowledge that it doesn't really accept any type, since there are some types for which it will raise an error), or would you prefer that I add a `#type: ignore[override]` comment to suppress MyPy errors specifically relating to the LSP on that line?

Reply:
Here's one way to please nypy and satisfy LSP:

```python
    def __contains__(self, other: object) -> bool:
        try:
            assert isinstance(other, Iterable)
            x, y = other
            assert isinstance(x, int)
            assert isinstance(y, int)
        except Exception:
            raise TypeError(
                "Dimensions.__contains__ requires an iterable of two integers"
            )
        width, height = self
        return width > x >= 0 and height > y >= 0
```

Reply:
Reducing the use of `Any` is always good

Reply:
That still breaks LSP! ["New exceptions cannot be thrown by the methods in the subtype, except if they are subtypes of exceptions thrown by the methods of the supertype." ](https://en.wikipedia.org/wiki/Liskov_substitution_principle). `tuple.__contains__`, the method of the supertype that we're here overriding, would never throw `TypeError` for anything. The only difference is, because we're lying in the type annotation (we're telling MyPy that the function accepts any subtype of `object`, whereas in fact it throws an exception for anything except `Iterable[int]`), MyPy can no longer detect the LSP violation.

Reply:
Oh, right.

But we can go around that by returning `NotImplemented` instead. That will throw a `TypeError` anyway.

Reply:
I think that would follow the letter of the LSP. But I'm still not sure it would follow the spirit of the LSP, given that we'd still *end up* with an exception being raised where none would be raised in the same method of the supertype. (Are there any situations where `tuple.__contains__` would return `NotImplemented`, leading to a `TypeError` being raised?) And we'd get a generic error message rather than the custom error message we get currently.

I think it's just a bit of a fool's errand tying ourselves in knots to try to conform to the LSP in this instance. (But just my opinion! 🙂)

Reply:
You know what, you're right. I think it's worth being pragmatic here. Let's return to an Iterable of ints, as you originally suggested. And keep that TypeError in, because it could cause errors. Silence any errors that come of it.